src/pages/AuthenticatePage/index.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+// import { Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { login } from '../../reducers/AuthenticationReducer/actions';
+
+import  './styles.css';
+
+class AuthenticatePage extends Component {
+  constructor(props) {
+    super();
+    this.state = { key: '' }
+  };
+
+  saveKey = (e) => {
+    this.setState({ key: e.target.value });
+  };
+
+  authenticate = (e) => {
+    e.preventDefault();
+    this.props.login(this.state.key).then(() => {
+      if (!this.props.error) {
+        this.props.history.push("/");
```

Comment:
better use setting of state + redirect component, a more proper way than programmatically pushing